packages/components/multiselect/src/Multiselect.tsx

```diff
@@ -86,14 +86,48 @@ export interface MultiselectProps extends CommonProps {
      * Use this prop to get a ref to the list of items of the component
      */
     listRef?: React.Ref<HTMLUListElement>;
-  };
+  } & Pick<CommonProps, 'className'>;
 
   /**
    * Function called when the popover loses its focus.
    */
   onBlur?: () => void;
 }
 
+// Scan through the whole hierachy until `filter` returns true and apply `transform`
+// Inspired from https://stackoverflow.com/a/70676868/17269164
+const iterateOverChildren = (
+  children: React.ReactNode,
+  filter: (child: React.ReactElement) => boolean,
+  transform: (child: React.ReactElement) => React.ReactElement,
+): React.ReactNode => {
+  return React.Children.map(children, (child) => {
+    // equal to (if (child == null || typeof child == 'string'))
+    if (!React.isValidElement(child)) return child;
+    if (!filter(child))
+      return iterateOverChildren(child.props.children, filter, transform);
+    return transform(child);
+  });
+};
+
+// Scan through the whole hierachy to count the number of children where `filter` returns true
+const countMatchingChildren = (
```

Comment:
i m not 100% sold on not allowing people to show other children than options. We already had the case for autocomplete where someone wanted to show that the backend just did not respond anything to show without a search.

so maybe if we have children but none is an option we just show that as content but not inside the multiselect item list?

Reply:
i think we are talking about different things.
When I interpret this piece of code correctly, if you do not hand over an MultiselectOption, none of the children gets rendered.
But i want to keep the possibility to render an individual message that is different from the "no matches found" information, that should only be rendered when the search returned no results.